ambry-api/src/main/java/com.github.ambry/network/NetworkSend.java

```diff
@@ -59,7 +51,7 @@ public Send getPayload() {
 
   public void onSendComplete() {
     if (metrics != null) {
-      metrics.updateResponseSendTime(SystemTime.getInstance().milliseconds() - sendStartTimeInMs);
+      metrics.updateSendTime(time.milliseconds() - sendStartTimeInMs);
```

Comment:
Why is this even `updateSendTime()`? Since `NetworkSendMetrics` is a black box to `NetworkSend`, this should probably be `metrics.onSendComplete()`. It even looks like the `NetworkRequestMetrics` object should contain the start time as well. Depending on the use case, it should either be provided the start time, or `NetworkSend` should call a method `metrics.onSendStart()` (looking at the code it seems like the former is good enough).